internal/httputil/mux.go

```diff
@@ -29,13 +30,22 @@ func GetPathRegexp(pattern string) (string, error) {
 			return "", err
 		}
 
+		// mux actually doesn't support ending $
+		// fix it to provide a regexp that's expected
+		if strings.HasSuffix(result, "\\$") {
+			result = result[:len(result)-2] + "$"
+		}
+
 		pathRegexpCache.Set(pattern, result)
 		return result, nil
 	}
 
 	if strings.HasPrefix(pattern, "/") {
```

Comment:
This is the breaking change, if a `/path` is defined, rather than `^/path`, it would wildcard match any url that contains path, anywhere in the url.